### ISSUE TITLE
archive.extracted fails on second run

### ISSUE THREAD

**Issue by UtahDave** [saltstack/salt#41540]
### Description of Issue/Question
`archive.extracted` fails the second time it is run.

### Setup
```
root@elsapo:/srv/salt# cat /srv/salt/testzip.sls 
test_zip:
  archive.extracted:
    - name: /tmp/testzip
    - source: salt://test.zip
    - source_hash: md5=8c4c6b8bf1b6a1834a8e33f8e04ed94f
    - archive_format: zip
```
### Steps to Reproduce Issue
Run the following 2 times and it successfully works the first time and I get the following error the second time:

```
[boucha@elsapo tmp]$ sudo salt \* state.sls testzip
dasalt:
----------
          ID: test_zip
    Function: archive.extracted
        Name: /tmp/testzip
      Result: False
     Comment: The below paths (relative to /tmp/testzip/) exist, but are the incorrect type (file instead of directory, symlink instead of file, etc.). To proceed with extraction, set 'force' to True. Note that this will remove these paths before extracting.
              
              - testdir/
     Started: 17:41:47.883112
    Duration: 50.946 ms
     Changes:   

Summary for dasalt
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
Total run time:  50.946 ms
ERROR: Minions returned with non-zero exit code
```

### Versions Report

```[boucha@elsapo tmp]$ salt --versions-report
Salt Version:
           Salt: 2016.11.3
 
Dependency Versions:
           cffi: 1.9.1
       cherrypy: unknown
       dateutil: 2.6.0
          gitdb: 0.5.4
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8.1
        libgit2: Not Installed
        libnacl: 1.4.4
       M2Crypto: 0.21.1
           Mako: 0.9.1
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: 2.17
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.6 (default, Oct 26 2016, 20:30:19)
   python-gnupg: 0.3.7
         PyYAML: 3.12
          PyZMQ: 14.0.1
           RAET: Not Installed
          smmap: 0.8.2
        timelib: Not Installed
        Tornado: 4.3
            ZMQ: 4.0.5
 
System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 3.13.0-119-generic
         system: Linux
        version: Ubuntu 14.04 trusty
```

Same result on 2016.11.5

Another user was getting another error when using `saltenv` on the cli.
```
salt 'minionid' state.sls testzip saltenv=someotherenv
```

**Comment by rickh563** [comment]
ZD-1490

**Comment by garethgreenaway** [comment]
@UtahDave I ran a quick test of this and got the expected results.

```
 gareth@blackoil  /tmp  sudo salt-call state.sls test.archive
local:
----------
          ID: test_zip
    Function: archive.extracted
        Name: /tmp/testzip
      Result: True
     Comment: All files in archive are already present
     Started: 10:30:58.847174
    Duration: 27.792 ms
     Changes:   

Summary for local
------------
Succeeded: 1
Failed:    0
------------
Total states run:     1
Total run time:  27.792 ms
```

Is there something different about the contents of the zipfile?

**Comment by UtahDave** [comment]
The zip file does have a subdirectory. I'm not sure what else could be different.

**Comment by garethgreenaway** [comment]
Maybe something quirky with the version of the zip binaries on 14.04.  Is it possible to provide a scrubbed version of the zip file in question for testing?

**Comment by UtahDave** [comment]
Yeah. I'll put one together that I can share.

**Comment by garethgreenaway** [comment]
Did some digging into this and found that the root of the issue is that the directory inside the zipfile is being flagged as a file by archive.list.  Once it's extracted it checks all existing files and because the directory inside is a directory and not simply a file, the test fails.  Digging more into why it's being flagged as a file.

**Comment by garethgreenaway** [comment]
This issue should be considered resolved and the fix is available on 2016.11. If the issue is not considered resolved or persists, please feel free to re-open this issue.
